packages/cli/test/e2e/cypress/integration/build.ts

```diff
@@ -0,0 +1,65 @@
+const BLITZ_PATH = "../blitz/bin/blitz"
+const TEST_APP_NAME = "app"
+
+// waiting 2 minutes for the command to execute or test breaks
+const execConfig = {
+  log: true,
+  timeout: 120000,
+}
+
+const buildAppCommand = (
+  appName: string = TEST_APP_NAME,
+  packageManager: "yarn" | "npm",
+  template: "full" | "minimal",
+  formLib: "react-final-form" | "react-hook-form" | "formik",
+  language: "typescript" | "javascript",
+): void => {
+  cy.exec(
+    `${BLITZ_PATH} new ${appName} --${packageManager} --template=${template}  --form=${formLib} --language=${language}`,
+    execConfig,
+  )
+  cy.writeFile(
+    `./${TEST_APP_NAME}/types.ts`,
+    "declare module 'react' { interface StyleHTMLAttributes<T> extends React.HTMLAttributes<T> { jsx?: boolean; global?: boolean }}",
+    {encoding: "utf8", flag: "a+"},
+  )
+
+  cy.readFile("../blitz/package.json").then((packageJson) => {
+    cy.exec(
+      `cd ${appName} && ${packageManager} add blitz@${packageJson.dependencies["@blitzjs/cli"]} --save`,
+    )
+    cy.readFile(`./${appName}/package.json`).then((testAppPackageJson) => {
+      expect(testAppPackageJson.dependencies["blitz"]).eqls(
+        packageJson.dependencies["@blitzjs/cli"],
+      )
+    })
+  })
+
+  cy.exec(`cd ${appName};${packageManager};${packageManager} run build`, execConfig)
+}
+
+describe("blitz new command", () => {
+  beforeEach(() => {
+    cy.exec(`rimraf ${TEST_APP_NAME}`)
+  })
+
+  afterEach(() => {
+    cy.exec(`rimraf ${TEST_APP_NAME}`)
+  })
+
+  it("Should create a new app", () => {
+    buildAppCommand(TEST_APP_NAME, "yarn", "full", "react-hook-form", "typescript")
+    cy.writeFile(
+      `./${TEST_APP_NAME}/.env`,
+      'DATABASE_URL="file:./db.sqlite\nSESSION_SECRET_KEY="ThisIsAKeyForTestingProductionBuildStart"',
+      {encoding: "utf8", flag: "a+"},
+    )
+    cy.exec(`cd ${TEST_APP_NAME} && yarn add pm2`)
+    cy.exec(`cd ${TEST_APP_NAME} && pm2 --name testBuild start yarn -- start`)
```

Comment:
why is pm2 being used? 

A test command of `blitz build && blitz start` should be sufficient, right?

Reply:
The main reason behind installing pm2, is the fact that `blitz start` does not run as a detached process from the command line. why I need to run a detached process ? to pass the step of starting the built server app and continue my tests down bellow. 

Cypress documentation gives an indication that running the server inside cypress is considered as an anti-pattern [see here](https://docs.cypress.io/guides/references/best-practices#Web-Servers). 

I think I took a wrong path imo, generating, building and starting  a new generated app could be done separately from outside cypress via a script or something. After the project is built and ready to go, we `yarn run start` the server,wait for the server till it starts and  after that run cypress to do the e2e test like singup/login..